arbnode/dataposter/leveldb/leveldb.go

```diff
@@ -60,8 +61,16 @@ func (s *Storage[Item]) GetContents(_ context.Context, startingIndex uint64, max
 	return res, it.Error()
 }
 
-func (s *Storage[Item]) GetLast(context.Context) (*Item, error) {
-	val, err := s.db.Get(lastItemKey)
+func (s *Storage[Item]) lastItemIdx(context.Context) ([]byte, error) {
+	return s.db.Get(lastItemIdxKey)
+}
+
+func (s *Storage[Item]) GetLast(ctx context.Context) (*Item, error) {
+	lastItemIdx, err := s.lastItemIdx(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting last item index: %w", err)
```

Comment:
I think this should return `nil, nil` if the last item index is missing (i.e. there are no items). We should also think about what happens when all items are pruned. In that case, we probably want to have pruning remove the last item index from the db.

Reply:
Done. Checking length, if it's 0 then returning nil, nil. So no need to remove last item idx from db when pruning (for all public methods behavior will be exactly the same).